src/test/java/ibm/jceplus/junit/base/BaseTestECDSASignature.java

```diff
@@ -759,6 +768,40 @@ public void testPostiveSigBytes() throws Exception {
         }
     }
 
+    @ParameterizedTest
+    @ValueSource(strings = {"secp256r1", "secp384r1", "secp521r1"})
+    public void testECDSASignatureWithInvalidKeySpec(String curveName) throws Exception {
+        ECPrivateKey ecPrivKey = makePrivateKey(curveName);
+
+        // Check using the private key for creating a digital signature
+        Signature sig = null;
+        KeyAgreement ka = null;
+        try {
+            sig = Signature.getInstance("SHA256withECDSA", this.getProviderName());
+            sig.initSign(ecPrivKey);
+            throw new RuntimeException("Expected exception for " +
+                    "ECDSA/" + sig.getAlgorithm() + "/" + curveName +
+                    " not thrown.");
+        } catch (InvalidKeyException ike) {
+            // We are expecting this to be caught
+            System.out.println("Caught expected exception for " +
+                    "ECDSA/" + sig.getAlgorithm() + "/" + curveName +
+                    ": " + ike);
+        }
+
+        // Next, try starting a ECDH operation
+        try {
```

Comment:
Should this be here? This is a `ECDH` test.

Reply:
removed, duplicated test in `BaseTestECDHKeyAgreement`